src/operator/tensor/elemwise_binary_scalar_op_basic.cc

```diff
@@ -53,16 +53,16 @@ static bool BinaryScalarStorageTypeWithDenseResultStorageType(const NodeAttrs& a
                                                               std::vector<int>* in_attrs,
                                                               std::vector<int>* out_attrs)  {
   bool dispatched = false;
+  const double alpha = nnvm::get<double>(attrs.parsed);
   if (common::ContainsOnlyStorage(*in_attrs, kDefaultStorage)) {
     dispatched = storage_type_assign(&out_attrs[0],
-                                     kDefaultStorage,
-                                     dispatch_mode,
-                                     DispatchMode::kFCompute);
-  } else if (dev_mask == kCPU) {
+      kDefaultStorage, dispatch_mode, DispatchMode::kFCompute);
+  } else {
```

Comment:
we should replace `else` with  `if (!dispatched && (instype == 'csr' || 'row_sparse'))`  to be explicit
There will be other stype such as MKL added soon